### ISSUE TITLE
Snowflake Tag Resource Should Support all Object Types

### ISSUE THREAD

**Issue by Willybaker1988** [Snowflake-Labs/terraform-provider-snowflake#1862]
### Is your feature request related to a problem? Please describe.

The existing resource available [snowflake_tag](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/tag) only supports assigning a tag to a database and schema, this is a limitation as not all tags should be assigned to these Snowflake objects.

### Describe the solution you'd like

There is a similar feature improvement mentioned here about default tags which would be useful https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/951.

The Snowflake documentation allows tags to be assigned **directly** to other Snowflake objects listed here https://docs.snowflake.com/en/user-guide/object-tagging#supported-objects.

The additional resource [snowflake_tag_association ](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/tag_association) is also limited by the scope of the [snowflake_tag](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/tag) being associated to a database and schema first.

An example in psuedo 

```
##table
resource "snowflake_tag" "tag" {
 name           = "cost_center"
  object_identifier {
    name     = snowflake_table.test.name
    database = snowflake_database.test.name 
    schema   = snowflake_schema.test.name
  }
  object_type = "TABLE"
  tag_value   = "engineering"
}
```

```
##role
resource "snowflake_tag" "tag" {
 name           = "cost_center"
  object_identifier {
    name     = snowflake_role.test.name
    database = null 
    schema   = null
  }
  object_type = "ROLE"
  tag_value   = "engineering"
}
```

**Comment by sfc-gh-jmichalak** [comment]
Hi @Willybaker1988 👋 

We've released a new v0.98.0 version ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.98.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0970--v0980)) with missing object types.

**Comment by sfc-gh-jmichalak** [comment]
Closing due to inactivity.

Also, we've just released v0.100.0 ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.100.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0990--v01000)) with more improvements in tag associations. Please upgrade to this version and let us know if you have any issues.
